pkg/registry/generic/registry/store.go

```diff
@@ -387,6 +440,39 @@ var (
 	errDeleteNow       = fmt.Errorf("delete now")
 )
 
+// return if we need to update the finalizers of the object
+func shouldUpdateFinalizers(accessor meta.Object, options *api.DeleteOptions) bool {
```

Comment:
suggest calling this "newFinalizersRequired" and making it return a list of finalizers that are needed.